src/library_signals.js

```diff
@@ -124,6 +124,22 @@ var funs = {
     ___setErrNo(ERRNO_CODES.EINTR);
     return -1;
   },
+#if ASSERTIONS
+  $siglongjmpWarningShown: false,
+  siglongjmp__deps: ['$siglongjmpWarningShown', 'longjmp'],
+  siglongjmp: function(env, value) {
+    // We cannot wrap the sigsetjmp, but I hope that
+    // in most cases siglongjmp will be called later.
+    if (!siglongjmpWarningShown) {
```

Comment:
Thanks, switched to `Runtime.warnOnce` that looks to be equally fast as my hand-written code.